pkg/validatingwebhook/repositoryserver_webhook.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validatingwebhook
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+const errValidate = "Validation Error"
+
+type RepositoryServerWebhook struct {
+}
+
+var _ webhook.CustomValidator = &RepositoryServerWebhook{}
+
+// TODO(user): change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
```

Comment:
is this source file generated?

Reply:
yes using kubebuilder, I have tweaked it for our use